jwst/mrs_imatch/mrs_imatch_step.py

```diff
@@ -328,6 +329,31 @@ def _match_models(models, channel, degree, center=None, center_cs='image'):
     #if np.isnan(sigma_data).any():
     #    print('a nan exists in sigma data')
 
+    # MRS fields of view are small compared to source sizes,
+    # and undersampling produces significant differences
+    # in overlap regions between exposures.
+    # Therefore use sigma-clipping to detect and remove sources
+    # (and unmasked outliers) prior to doing the background matching.
+    # Loop over input exposures
+    for i in range(0,len(models)):
+        image = image_data[i]
+        mask = mask_data[i]
+        weight = sigma_data[i]
+        # Do statistics wavelength by wavelength
+        for j in range(0,len(image)):
+            thisimg = image[j,:,:]
+            thismask = mask[j,:,:]
+            thiswght = weight[j,:,:]
+            # Sigma clipped statistics, ignoring zeros where no data
+            _, themed, clipsig = sigclip(thisimg, mask_value=0.)
+            im1d = thisimg.ravel()
+            mask1d = thismask.ravel()
+            wght1d = thiswght.ravel()
+            # Reject beyond 3sigma
+            reject = np.where(np.abs(im1d - themed) > 3*clipsig)
```

Comment:
I think it is important to have the value of "number of sigmas" (3 in this case) be settable for the step. For example, add a new parameter `nsig(ma)`. But, if you think that for now it would be better not to introduce new parameters to the step, then leave it as is.

Reply:
I'd been thinking about whether that was worthwhile doing now or delaying until sometime later; I'll test that out along with your other suggestions above (was sure you'd have some thoughts on how to make my clunky code more elegant!)

Reply:
You know, maybe it is better not to do this now. Let's wait until there is need for this. It is more difficult to remove a parameter than to add one. Also add spaces around `*` in `3 * clipsig`.